src/stormpath.social-login.google.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+(function() {
+  function createGoogleScriptTag(id, onLoad) {
+    var jsElement = document.createElement('script');
+
+    jsElement.id = id;
+    jsElement.src = '//apis.google.com/js/api:client.js';
+    jsElement.onload = onLoad;
+
+    return jsElement;
+  }
+
+  function loadGoogleSDK(onLoad) {
+    var id = 'google-jssdk';
+    var firstJsElement = document.getElementsByTagName('script')[0];
+    var jsElement = createGoogleScriptTag(id, onLoad);
+
+    if (document.getElementById(id)) {
+      onLoad();
+      return;
+    }
+
+    firstJsElement.parentNode.insertBefore(jsElement, firstJsElement);
+  }
+
+  angular.module('stormpath.googleLogin', [])
+  .provider('$googleLogin', function() {
+    this.$get = ['$window', '$q', function googleLoginFactory($window, $q) {
+      var _clientId;
+      var auth2;
+
+      function GoogleLoginService() {
+      }
+
+      GoogleLoginService.prototype.name = 'Google';
+
+      GoogleLoginService.prototype.clientId = function clientId(value) {
```

Comment:
I think this method should be named `setClientId`.  It also feels not-DRY to be repeating this function in every provider service.